Src/java/cql-to-elm/src/test/java/org/cqframework/cql/cql2elm/GenericOverloadsTests.java

```diff
@@ -0,0 +1,112 @@
+            for (ExpressionDef def : library.getStatements().getDef()) {
+                defs.put(def.getName(), def);
+            }
+        }
+        return library;
+    }
+
+
+    private static CqlTranslator getTranslator(boolean enableResultTypes, SignatureLevel level) throws IOException {
+        var options = new CqlCompilerOptions();
+        options.getOptions().clear();
+        options.setSignatureLevel(level);
+        if (enableResultTypes) {
+            options.getOptions().add(CqlCompilerOptions.Options.EnableResultTypes);
+        }
+
+        return TestUtils.createTranslator(CQL_TEST_FILE, options);
+    }
+
+    private List<FunctionDef> stringifies(Library library) {
+        return library.getStatements().getDef()
+            .stream()
+            .filter(x -> "Stringify".equals(x.getName()))
+            .filter(FunctionDef.class::isInstance)
+            .map(FunctionDef.class::cast)
+            .collect(toList());
+    }
+
+    private void validateResultTypes(FunctionDef functionDef) {
+        assertEquals(2, functionDef.getOperand().size());
+
+        var operand = functionDef.getOperand().get(0);
+        assertThat(operand.getOperandTypeSpecifier(), instanceOf(ListTypeSpecifier.class));
+        var listSpecifier = (ListTypeSpecifier)operand.getOperandTypeSpecifier();
+        assertThat(listSpecifier.getElementType(), instanceOf(NamedTypeSpecifier.class));
+        var namedSpecifier = (NamedTypeSpecifier)listSpecifier.getElementType();
+        assertNotNull(namedSpecifier.getName());
+        assertNotNull(namedSpecifier.getResultType());
+
+        var second = functionDef.getOperand().get(1);
+        assertThat(second.getOperandTypeSpecifier(), instanceOf(ListTypeSpecifier.class));
+        listSpecifier = (ListTypeSpecifier)operand.getOperandTypeSpecifier();
+        assertThat(listSpecifier.getElementType(), instanceOf(NamedTypeSpecifier.class));
+        namedSpecifier = (NamedTypeSpecifier)listSpecifier.getElementType();
+        assertNotNull(namedSpecifier.getName());
+        assertNotNull(namedSpecifier.getResultType());
+    }
+
+    @Test
+    public void TestResultTypes() throws IOException {
+        Library library = getLibrary(true, SignatureLevel.Overloads);
+
+        var stringifies = stringifies(library);
+        stringifies.forEach(this::validateResultTypes);
+    }
+
+    @Test
+    public void TestNoResultTypes() throws IOException {
+        Library library = getLibrary(false, SignatureLevel.Overloads);
+
+        var stringifies = stringifies(library);
+        stringifies.forEach(this::validateResultTypes);
+    }
+
+    @Test
+    public void TestResultTypesSignatureNone() throws IOException {
+        Library library = getLibrary(true, SignatureLevel.None);
+
+        var stringifies = stringifies(library);
+        stringifies.forEach(this::validateResultTypes);
+    }
+
+    @Test
+    public void TestNoResultTypesSignatureNone() throws IOException {
```

Comment:
For completeness I suggest doing the same for the other 2 signature types, and consider doing this as part a parameterized test.